### PR TITLE
Adding pytest integration

### LINKED ISSUE #1260: [FEAT][USE] PyTest userguide and usecase
Write a pytest plugin to enable the checks to run as unit tests. Maybe the plugin itself is not needed, but we want to demonstrate how deepchecks can be used as part of the project’s unit tests.

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Closes #1260 

image

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

**Review by matanper — APPROVED** [review]

File: docs/source/user-guide/integrations/pytest.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+======
+pytest
+======
+
+This tutorial demonstrates how deepchecks can be used inside unit tests performed on data or model, with the pytest
+framework.
+We will use the ``diabetes`` dataset from scikit-learn, and check whether certain columns does not contain drift
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/pytest.rst:7]:
```suggestion
We will use the ``diabetes`` dataset from scikit-learn, and check whether certain columns contain drift
```

File: docs/source/user-guide/integrations/pytest.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @pytest.fixture(scope='session')
+    def diabetes_df():
+        diabetes = load_diabetes(return_X_y=False, as_frame=True).frame
+        return diabetes
+
+Implementing the Test
+-----------------------
+
+Now, we will implement a test that will check if some columns in the dataset are not drifting between the training and
```
Comment (noamzbr) [inline docs/source/user-guide/integrations/pytest.rst:36]:
```suggestion
Now, we will implement a test that will check if some columns in the dataset have drifted between the train and test datasets.
```

**Review by noamzbr — APPROVED** [review]
Approved with minor suggestions.

File: docs/source/user-guide/integrations/pytest.rst (hunk 1 above)
Comment (shir22) [reply docs/source/user-guide/integrations/pytest.rst:7]: +1

File: docs/source/user-guide/integrations/pytest.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+======
+pytest
```
Comment (shir22) [inline docs/source/user-guide/integrations/pytest.rst:2]:
```suggestion
Pytest
```

File: docs/source/user-guide/integrations/pytest.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+.. code-block:: python
+
+    import pytest
+    from sklearn.datasets import load_diabetes
+    from sklearn.model_selection import train_test_split
+
+    from deepchecks import Dataset
+    from deepchecks.tabular.checks import TrainTestFeatureDrift
+
+Defining PyTest Fixtures
```
Comment (shir22) [inline docs/source/user-guide/integrations/pytest.rst:19]:
```suggestion
Defining Pytest Fixtures
```

**Review by shir22 — APPROVED** [review]
:)

**Pull request merged**